src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -451,6 +486,31 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
         return 0;
     }
 
+    static winrt::event<NewConnectionHandler> _newConnectionHandlers;
+
+    winrt::event_token ConptyConnection::NewConnection(NewConnectionHandler const& handler) { return _newConnectionHandlers.add(handler); };
+    void ConptyConnection::NewConnection(winrt::event_token const& token) { _newConnectionHandlers.remove(token); };
+
+    HRESULT ConptyConnection::NewHandoff(HANDLE in, HANDLE out, HANDLE signal, HANDLE process) noexcept
+    try
+    {
+        auto conn = winrt::make<ConptyConnection>((uint64_t)signal, (uint64_t)in, (uint64_t)out, (uint64_t)process);
+        _newConnectionHandlers(conn);
+
+        return S_OK;
+    }
+    CATCH_RETURN()
+
+    void ConptyConnection::StartInboundListener()
+    {
+        THROW_IF_FAILED(CTerminalHandoff::s_StartListening(&ConptyConnection::NewHandoff));
+    }
+
+    void ConptyConnection::StopInboundListener()
+    {
+        THROW_IF_FAILED(CTerminalHandoff::s_StopListening());
```

Comment:
should this take down the app?

Reply:
probably not but it's not called anymore because I took the action out and just remains for symmetry.

Also the caller could catch it?

Do you want me to catch the startup call in `TerminalPage::_OnFirstLayout`?

Reply:
the on first layout one is now called with a fail fast if failed because a tabless terminal is a super weird state and the crash will notify com that something went wrong. also left a comment to this effect.

this side on stopping is unused and I will refine it further in #9475